rebaser.go

```diff
@@ -137,6 +142,71 @@ func (r *Rebaser) validateRebaseable(appImg imgutil.Image) error {
 		if !r.Force && rebaseable == "false" {
 			return fmt.Errorf("app image is not marked as rebaseable")
 		}
+
+		// check the OS, architecture, and variant values
+		// if they are not the same, the image cannot be rebased unless the force flag is set
+		if !r.Force {
+			appOS, err := appImg.OS()
```

Comment:
`platform.GetTargetFromImage(image imgutil.Image)` could make this a bit simpler, maybe

Reply:
Thanks for that! Seems to have cleaned up things in f4092ea5